### PR TITLE
CB-15392 add flow to handle ephemeral datahubs to pick up changes from SDX resize operation

### PR DESCRIPTION
**Pull request by yg-gd**
**Here is the summary of the change**
1. Created two flows, one in cloudbreak to trigger cluster refresh the changes and another from datalake to wait on the CB flow.
2. Made changes to avoid stopping DH when DL is resized.
Added [CB-16637](https://jira.cloudera.com/browse/CB-16637) for E2E test.

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/MockSdxResizeTests.java [hunk 1]
```diff
@@ -72,4 +78,38 @@ public void testDefaultSDXResizeSuccessfully(MockedTestContext testContext) {
 //                .validate();
                 //TODO Make sure that the runtime is same, storage location is also same.
     }
+
+    //@Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
```
Comment (foldik) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/MockSdxResizeTests.java:82]: Can you enable this test?

**Review by foldik — CHANGES_REQUESTED** [review]

**Comment by yg-gd** [comment]
@kkalvagadda1 could you ensure that ClouderaManagerClientActions.getCmServiceResourceHiveHost is checking the correct thing?

**Review by foldik — APPROVED** [review]
LGTM

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]

**Review by kkalvagadda1 — APPROVED** [review]

**Review by foldik — APPROVED** [review]
LGTM

**Review by sxxgrc — APPROVED** [review]

**Review by gregito — CHANGES_REQUESTED** [review]
small and fast requests left

File: core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterServicesRestartService.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        configs.put(JDBC_URL_OVERRIDE, hiveRdsView.getConnectionURL());
+        return configs;
+    }
+
+    private boolean isDatalakeCreatedAfterDataHub(Stack datalakeStack, Stack dataHubStack) {
+        return datalakeStack.getCreated() > dataHubStack.getCreated();
+    }
+
+    private boolean shouldReloadDatabaseConfig(CmTemplateProcessor blueprintProcessor) {
+        return blueprintProcessor.isCMComponentExistsInBlueprint("HIVEMETASTORE");
```
[the remark this answers is not in the document]
Comment (kkalvagadda1) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterServicesRestartService.java:135]: adddressed.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterServicesRestartService.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    private void updateDatabaseConfiguration(Stack datalakeStack, Stack dataHubStack, String service, DatabaseType databaseType) {
+        Cluster cluster = clusterService.getById(datalakeStack.getCluster().getId());
+        Optional<RDSConfig> rdsConfig = postgresConfigService.createRdsConfigIfNeeded(datalakeStack, cluster, databaseType)
+                .stream().filter(config -> config.getType().toLowerCase().equals(databaseType.toString().toLowerCase()))
+                .findFirst();
+        try {
+            if (rdsConfig.isPresent()) {
+                apiConnectors.getConnector(dataHubStack).updateServiceConfig(service, getRdsConfigMap(rdsConfig.get()));
```
[the remark this answers is not in the document]
Comment (kkalvagadda1) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterServicesRestartService.java:108]: addressed.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterServicesRestartService.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        apiConnectors.getConnector(stack).startClusterMgmtServices();
+        return refreshClusterOnRestart(stack, datalakeStack, blueprintProcessor);
+    }
+
+    public int refreshClusterOnRestart(Stack stack, Stack datalakeStack, CmTemplateProcessor blueprintProcessor) throws CloudbreakException {
+        LOGGER.info("Triggering update of remote data context");
+        clusterBuilderService.configureManagementServices(stack.getId());
+        if (shouldReloadDatabaseConfig(blueprintProcessor)) {
+            //Update Hive service database configuration
+            updateDatabaseConfiguration(datalakeStack, stack, HIVE_SERVICE, DatabaseType.HIVE);
```
[the remark this answers is not in the document]
Comment (kkalvagadda1) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterServicesRestartService.java:94]: addressed.

**Pull request merged**